### ISSUE TITLE
Functionality to look up account IDs based on account name

### ISSUE THREAD

**Issue by vongkim** [awslabs/landing-zone-accelerator-on-aws#393]
**Describe the bug**
The documentation states that you can reference an account ID using the following syntax:

```
  parameters:
   - name: TrustedAccounts
     value: {{account Management}}
```

This only works if there is a replacement variable in the same configuration. e.g `{{AReplacementVariable}}` present within the configuration.

**To Reproduce**
In the `customizations-config.yaml` declare the following variable:

```
auditAccountId: '{{account Audit}}'
```

This will result in the following error:

```
2024-02-05 17:00:51.456 | error | accounts-config | Account ID not found for Audit. Validate that the emails in the parameter ManagementAccountEmail of the AWSAccelerator-InstallerStack and account configs (accounts-config.yaml) match the correct account emails shown in AWS Organizations.
```

However, if you have a replacement variable present, it will work e.g

```
auditAccountId: '{{account Audit}}'
example: {{example}}
```

I've narrowed it down to the following code in `source/packages/@aws-accelerator/accelerator/lib/config-validator.ts`

```
const props = {
  ...
  replacementsPresent: areReplacementsPresent(),
};

...

function areReplacementsPresent() {
  // Matches lookup values excluding account lookups such as {{account Management}}
  // Account lookups do not require existence of replacements-config.yaml file
  const regex = new RegExp('{{(?!.*(account )).*}}');
  ...
}


async function validateConfig(props: {
  partition: string;
  region: string | undefined;
  enableSingleAccountMode: boolean;
  account: string | undefined;
  replacementsPresent: boolean;
}) {
  ...
  // Load accounts config
  let accountsConfig: AccountsConfig | undefined = undefined;
  try {
    accountsConfig = AccountsConfig.load(configDirPath);
    if (props.replacementsPresent) {
      await accountsConfig.loadAccountIds(props.partition, props.enableSingleAccountMode, orgsEnabled, accountsConfig);
    }
  } catch (e) {
    initErrors.push({ file: AccountsConfig.FILENAME, message: e });
  }
  ...
}

```

The `areReplacementsPresent` function will return false when the configuration file only references an account unless there is a replacement variable present in the same configuration file as stated above. This results in the `validateConfig` function to not call the `loadAccountIds` function.

**Expected behavior**
The `loadAccountIds` is called and configuration files can reference account IDs by name regardless of having used regular replacement variables.

**Please complete the following information about the solution:**

- [x] Version: v1.6.0
- [x] Region: ap-southeast-2
- [x] Was the solution modified from the version published on this repository? No
- [x] If the answer to the previous question was yes, are the changes available on GitHub? N/A
- [x] Have you checked your [service quotas](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html) for the services this solution uses? N/A
- [x] Were there any errors in the CloudWatch Logs? N/A

**Comment by ardens-jw** [comment]
@vongkim just wanted to put another voice behind this - came across this bug yesterday and was going to raise my own issue for it. 

The comment `// Matches lookup values excluding account lookups such as {{account Management}}` seems to suggest that somewhere else in the code this match is performed but I don't think there is.

As a workaround for now, I've amended the regex in `config-validator.ts` 

**Before**
```typescript
const regex = new RegExp('{{(?!.*(account )).*}}');
```

**After**
```typescript
const regex = new RegExp('{{.*(?:account\\s)?\\w+\\s*}}');
```

**Comment by erwaxler** [comment]
Hello @vongkim , thank you for filing an issue with the Landing Zone Accelerator team! I can confirm I was able to replicate that behavior, the team has fixed this bug and it is slated to be incorporated in the next release.

I will keep this issue open until that release is available. Thanks again for supporting the Landing Zone Accelerator!

**Comment by erwaxler** [comment]
Hi @vongkim , this has been included in our [v1.7.0 release](https://github.com/awslabs/landing-zone-accelerator-on-aws/releases/tag/v1.7.0) so I will be closing this issue. Thanks for supporting the Landing Zone Accelerator!
